agent/proxycfg-glue/glue.go

```diff
@@ -141,6 +141,12 @@ func newUpdateEvent(correlationID string, result any, err error) proxycfg.Update
 	if acl.IsErrNotFound(err) {
 		err = proxycfg.TerminalError(err)
 	}
+	// these are also errors where we should mark them
+	// as terminal for the sake of proxycfg, since they require
+	// a resubscribe.
+	if err == stream.ErrSubForceClosed || err == stream.ErrShuttingDown {
+		err = proxycfg.TerminalError(err)
+	}
```

Comment:
Maybe worth using `errors.Is` here incase the errors are wrapped?

```suggestion
	if errors.Is(err, stream.ErrSubForceClosed) || errors.Is(err, stream.ErrShuttingDown) {
		err = proxycfg.TerminalError(err)
	}
```